### PR TITLE
One branch dhaval

### PR DESCRIPTION
**Pull request by dhaval2989**

### CODE REVIEW

File: src/app/layout/nova-sidenav/nova-sidenav.component.html [hunk 1]
```diff
@@ -5,15 +5,15 @@
             <mat-icon *ngIf="isExpanded" class="sidenav-icon" aria-label="Menu" (click)="closeSideNav()">menu</mat-icon>
```
Comment (Sina-pz) [inline src/app/layout/nova-sidenav/nova-sidenav.component.html:5]: please name of component change to nv-

File: src/app/layout/nova-sidenav/nova-sidenav.component.html [hunk 2]
```diff
@@ -5,15 +5,15 @@
...
             <h2 *ngIf="isExpanded" class="sidenav-txt color-main">MAIN MENU</h2>
         </div>
         <mat-nav-list>
             <!-- Itterrating all the menu items in the side menu -->
             <mat-list-item *ngFor='let item of sidenavItems; let i = index'>
                 <div class="hoverColor"></div>
-                <mat-icon mat-list-icon *ngIf='i!==8'><img class="sidenav-menuicon" src="{{item.icon}}"></mat-icon>
-                <mat-icon mat-list-icon *ngIf='i==8'><img class="sidenav-menuicon2" src="{{item.icon}}"></mat-icon>
+                <mat-icon mat-list-icon *ngIf='i!==8'><img class="sidenav-menuicon" src="{{item.icon}}" alt="menu1"></mat-icon>
+                <mat-icon mat-list-icon *ngIf='i==8'><img class="sidenav-menuicon2" src="{{item.icon}}" alt="menu2"></mat-icon>
```
Comment (Sina-pz) [inline src/app/layout/nova-sidenav/nova-sidenav.component.html:16]: if possible change the name to a descriptive name; not use 2

File: src/app/main/components/dashboard/dashboard.component.html [hunk 3]
```diff
@@ -1,98 +1,128 @@
...
 <app-nova-sidenav>
+  <h1 class="comments-header"><strong>Comments</strong></h1>
+  <app-comment-box [comments]="comments"></app-comment-box>
+  <br />
+  <app-nova-panel>
+    <app-nv-custom-table></app-nv-custom-table>
+  </app-nova-panel>
+  <hr />
+  <h1 class="comments-header"><strong>Star Rating</strong></h1>
+  <app-ratting [type]="'star'"></app-ratting>
```
Comment (Sina-pz) [inline src/app/main/components/dashboard/dashboard.component.html:12]: if possible define variable in ts file and use them here; e.g. using enum for different types

File: src/app/main/components/dashboard/dashboard.component.html [hunk 4]
```diff
@@ -1,98 +1,128 @@
...
+  <br />
+  <hr />
+  <h1 class="comments-header"><strong>Heart Rating</strong></h1>
+  <app-ratting [type]="'heart'"></app-ratting>
+  <br />
+  <hr />
+  <h1 class="comments-header"><strong>Sucess Progressive Bar</strong></h1>
+  <app-progressive-bar
+    [color]="'sucess'"
```
Comment (Sina-pz) [inline src/app/main/components/dashboard/dashboard.component.html:21]: same

File: src/app/main/components/dashboard/dashboard.component.ts [hunk 5]
```diff
@@ -4,51 +4,52 @@ import { Component, Input, OnInit } from '@angular/core';
...
   types = [{component:TextAreaComponent, name:'TabOne'}, {component:NovatekLogoComponent, name:'TabTwo'}, {component:TextAreaComponent, name:'TabThree'}];
   invertedTypes = [{component:NvSearchComponent, name:'ITabOne'}, {component:NovatekLogoComponent, name:'ITabTwo'}, {component:TextAreaComponent, name:'ITabThree'}, {component:NvSearchComponent, name:'ITabFour'}];
   leftTypes = [{component:NvSearchComponent, name:'lTabOne'}, {component:NovatekLogoComponent, name:'lTabTwo'}, {component:TextAreaComponent, name:'lTabThree'}];
   rightTypes = [{component:NovatekLogoComponent, name:'RTabOne'}, {component:NvSearchComponent, name:'RTabTwo'}, {component:TextAreaComponent, name:'RTabThree'}, {component:NvSearchComponent, name:'RTabFour'}];
   progressiveColor: string = '#198515';
-  progressiveStatus:string = '50%'; 
+  progressiveStatus: string = '50%';
   // mockData for header
   @Input() person: Person = mockProfileMenu();
   avtar = '../../../../assets/images/avtar.jpg';
```
Comment (Sina-pz) [inline src/app/main/components/dashboard/dashboard.component.ts:29]: we already have avatar component; if not mistaken you need to pass both url and name to use that one

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+    font-style: normal;
+    font-weight: 500;
+    font-size: 14px;
+    font-family: "Lato", "Roboto", "sans-serif";
+    line-height: 23px;
+    color: #5D5D5D;
+  }
+  .body:link {
+    line-height: 18px;
+    color: #1791F1;
```
Comment (Sina-pz) [inline src/app/main/components/user-list-view/user-list-view.component.scss:14]: please use base.scss

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+  }
+  .body:visited {
+    line-height: 18px;
+    color: #004463;
```
Comment (Sina-pz) [inline src/app/main/components/user-list-view/user-list-view.component.scss:18]: same

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+  ::ng-deep .search-placeholder-align .mat-form-field-infix {
+    position: relative;
+    top: -6px;
+  }
+  
+  ::ng-deep .search-placeholder-align .mat-form-field-label-wrapper {
+    margin-top: 4px;
+  }
+  
+  .padding-3 {
```
Comment (Sina-pz) [inline src/app/main/components/user-list-view/user-list-view.component.scss:74]: padding-3 has value of 4px, is it ok?

File: src/app/main/components/user-list-view/user-list-view.component.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        this.isLeftVisible = !this.isLeftVisible
+        break;
+      case '2':
+        this.showPanel='3';
+        break;
+      case '3':
+          this.showPanel='1';
+        break;
+    }
+    console.log('panel clicked', panel);
```
Comment (Sina-pz) [inline src/app/main/components/user-list-view/user-list-view.component.ts:31]: please delete

File: src/app/reusable/directives/drag-drop/drag-and-drop.directive.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import {Directive, EventEmitter, HostListener, Output} from '@angular/core';
```
Comment (Sina-pz) [inline src/app/reusable/directives/drag-drop/drag-and-drop.directive.ts:1]: we already have this directive!! now there are two similar directive

File: src/app/reusable/edit-user/edit-user.component.scss [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    width:100%;
+}
+
+.mr-tp-2{
+    margin-top: 2%;
+}
+
+.input-fld{
+    height: 40px;
+    border: 2px solid #DEDEDF;
```
Comment (Sina-pz) [inline src/app/reusable/edit-user/edit-user.component.scss:17]: please use base.scss

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.html [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    <div class="mr-tp-2">
+        <label>Company Logo</label>
+        <div class="files-container">
+           
+          <div class="container-upload margin-top-4 pd-5" appDragAndDrop
+               (file)="dragAndDropFile($event)" aria-label="upload files">
+            <input hidden type="file"  #fileInput (change)="uploadFile($event);">
+            <div class="d-flex flex-column text-center mt-1">
+              <div class="center-img-block margin-bottom-10">
+                <img class="img-file" height="35px" src="assets/images/doc.png" alt="pdf file logo">
```
Comment (Sina-pz) [inline src/app/reusable/nv-company-general-info/nv-company-general-info.component.html:15]: please set height in style

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.scss [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+  .pd-5{
+      padding: 5%;
+  }
+
+  .upload-btn{
+    width: 12%;
+    margin: 2% 0 0 45%;
+    background-color: $btn-color;
```
Comment (Sina-pz) [inline src/app/reusable/nv-company-general-info/nv-company-general-info.component.scss:69]: what is btn-color?

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+})
+export class NvCompanyGeneralInfoComponent implements OnInit {
+  file!: File;
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  uploadFile(event: any): void{
+    console.log('in upload', event.target.files[0].name);
```
Comment (Sina-pz) [inline src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts:16]: please delete

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  dragAndDropFile(file: any): void{
+    console.log('in drag', file.name);
```
Comment (Sina-pz) [inline src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts:20]: same

File: src/app/reusable/nv-custom-datatable/nv-custom-datatable.component.scss [hunk 16]
```diff
@@ -97,23 +95,22 @@ thead {
     background: hsl(240, 8%, 97%) !important;
 }
 tr.mat-header-row {
-    height: 40px !important;
     background-color: #fff !important;
```
Comment (Sina-pz) [inline src/app/reusable/nv-custom-datatable/nv-custom-datatable.component.scss:98]: same

File: src/app/reusable/slide-panel/slide-panel.component.ts [hunk 17]
```diff
@@ -0,0 +1,27 @@
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { animate, state, style, transition, trigger } from '@angular/animations';
+
+type PaneType = 'left' | 'right';
```
Comment (Sina-pz) [inline src/app/reusable/slide-panel/slide-panel.component.ts:4]: if possible put it in data-model or view-model (not exist yet)

File: src/app/reusable/user-list-table/user-list-table.component.scss [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+mat-paginator {
+  margin-bottom: 47px !important;
+}
+
+.search-box {
+  border: 1px solid #dededf;
+  margin: 1%;
+  border-radius: 4px;
+  height: 40px;
+  background: #fff;
```
Comment (Sina-pz) [inline src/app/reusable/user-list-table/user-list-table.component.scss:12]: base.scc

File: src/styles.scss [hunk 19]
```diff
@@ -3,7 +3,7 @@
 @import "./../src/app/reusable/theme/base.scss";
 
 html, body { height: 100%; }
-body { margin: 0; font-family: "Lato", Roboto, "Helvetica Neue", sans-serif;  overflow-y: hidden; }
+body { margin: 0; font-family: "Lato", Roboto, "Helvetica Neue", sans-serif; }
 
 .ngx-pagination .pagination-previous a::before, .ngx-pagination .pagination-previous.disabled::before {
```
Comment (Sina-pz) [inline src/styles.scss:8]: if possible change these style and put them into related component

File: src/app/main/components/dashboard/dashboard.component.html [hunk 20]
```diff
@@ -1,98 +1,128 @@
...
+    [size]="warningProgressivebarValue"
+    [progressiveTxt]="warningTxt"
+    [statusText]="textColor"
+  ></app-progressive-bar>
+  <br />
+  <hr />
+  <h1 class="comments-header"><strong>error Progressive Bar</strong></h1>
+  <app-progressive-bar
+    [color]="'error'"
+    [size]="SucessprogressiveBarValue"
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/dashboard/dashboard.component.html:39]: Follow camelCase

File: src/app/main/components/dashboard/dashboard.component.html [hunk 21]
```diff
@@ -1,98 +1,128 @@
...
+  <hr />
+  <div>
+    <nv-placeholder-Bigcard
+      [smallCardData]="smallCardData"
+    ></nv-placeholder-Bigcard>
+  </div>
 </app-nova-sidenav>
-<app-nova-footer></app-nova-footer>
\ No newline at end of file
+<app-nova-footer></app-nova-footer>
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/dashboard/dashboard.component.html:128]: Change to app-nv-footer.

File: src/app/main/components/dashboard/dashboard.component.ts [hunk 22]
```diff
@@ -4,51 +4,52 @@ import { Component, Input, OnInit } from '@angular/core';
...
 import { NvSearchComponent } from 'src/app/reusable/nv-search/nv-search.component';
-import { PlaceHolderCard } from 'src/app/model/placeHolderCard';
-import { PlaceHolderBanner } from 'src/app/model/placeHolderBanner';
-import { PlaceholderBigCard } from 'src/app/model/PlaceholderBigCard';
 import { Notification, Person } from 'src/app/model/data-model';
 
-
-
 @Component({
   selector: 'app-dashboard',
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/dashboard/dashboard.component.ts:14]: Change to app-nv-dashboard.

File: src/app/layout/nova-footer/nova-footer.component.html [hunk 23]
```diff
@@ -6,7 +6,7 @@
         <li class="lang-style">
```
Comment (veeranjaneyuludavuluri) [inline src/app/layout/nova-footer/nova-footer.component.html:6]: Change component name to nv-footer.

File: src/app/main/components/user-list-view/user-list-view.component.html [hunk 24]
```diff
@@ -0,0 +1,23 @@
+<app-nv-header ></app-nv-header>
+<app-nv-sub-menu></app-nv-sub-menu>
+<br>
+
+  <div class="row">
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.html:5]: Change name to app-nv-user-list-view.

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 25]
```diff
@@ -0,0 +1,317 @@
+/* The Design System's styles can be added here */
+ 
+   
+  .body {
+    font-style: normal;
+    font-weight: 500;
+    font-size: 14px;
+    font-family: "Lato", "Roboto", "sans-serif";
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.scss:8]: Is Lato working properly in this case?

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 26]
```diff
@@ -0,0 +1,317 @@
...
+    line-height: 18px;
+    color: #1791F1;
+  }
+  .body:visited {
+    line-height: 18px;
+    color: #004463;
+  }
+  .body:hover {
+    line-height: 18px;
+    color: #81BDED;
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.scss:22]: Use from base.scss if not add there and re use it here.

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 27]
```diff
@@ -0,0 +1,317 @@
...
+  }
+ 
+  
+  mat-paginator {
+    margin-bottom: 47px !important;
+  }
+  
+  .search-box {
+    border: 1px solid #dededf;
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.scss:31]: Use from base.scss if not add there and re use it here.

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 28]
```diff
@@ -0,0 +1,317 @@
...
+    margin: 1%;
+    border-radius: 4px;
+    height: 40px;
+    background: #fff;
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.scss:35]: Use from base.scss if not add there and re use it here.

File: src/app/main/components/user-list-view/user-list-view.component.scss [hunk 29]
```diff
@@ -0,0 +1,317 @@
...
+    color: #9a9a9a;
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.scss:36]: Use from base.scss if not add there and re use it here.

File: src/app/main/components/user-list-view/user-list-view.component.ts [hunk 30]
```diff
@@ -0,0 +1,34 @@
+import { Breakpoints } from '@angular/cdk/layout';
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-user-list-view',
```
Comment (veeranjaneyuludavuluri) [inline src/app/main/components/user-list-view/user-list-view.component.ts:5]: Add nv-

File: src/app/reusable/user-list-table/user-list-table.component.ts [hunk 31]
```diff
@@ -0,0 +1,199 @@
...
+
+  /**
+ * 
+ * @param event 
+ * This method is getting executed when user moves to any page using pagination
+ * dropDown selected item
+ */
+  onItemSelect(item: string): void {
+    if (item) {
+      console.log('selectedItem >>>> ' + +item);
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/user-list-table/user-list-table.component.ts:122]: Delete it.

File: package.json [hunk 32]
```diff
@@ -8,7 +8,8 @@
     "test": "ng test",
     "lint": "ng lint",
     "e2e": "ng e2e",
-    "compodoc": "npx compodoc -p tsconfig.doc.json -s -r 8081"
+    "compodoc": "npx compodoc -p tsconfig.doc.json -s -r 8081",
+    "sonar": "node sonar-project.js"
```
[the remark this answers is not in the document]
Comment (dhaval2989) [reply package.json:12]: Its working fine for me and Veera.

File: src/app/main/components/dashboard/dashboard.component.html (hunk 3 above)
Comment (dhaval2989) [reply src/app/main/components/dashboard/dashboard.component.html:12]: Its not part of my component.

File: src/app/main/components/dashboard/dashboard.component.ts (hunk 5 above)
Comment (dhaval2989) [reply src/app/main/components/dashboard/dashboard.component.ts:29]: Its not part of my component.

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 6 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:14]: fixed

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 7 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:18]: fixed

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 8 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:74]: fixed

File: src/app/main/components/user-list-view/user-list-view.component.ts (hunk 9 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.ts:31]: fixed

File: src/app/reusable/directives/drag-drop/drag-and-drop.directive.ts (hunk 10 above)
Comment (dhaval2989) [reply src/app/reusable/directives/drag-drop/drag-and-drop.directive.ts:1]: This is not part of my component

File: src/app/reusable/edit-user/edit-user.component.scss (hunk 11 above)
Comment (dhaval2989) [reply src/app/reusable/edit-user/edit-user.component.scss:17]: fixed

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.html (hunk 12 above)
Comment (dhaval2989) [reply src/app/reusable/nv-company-general-info/nv-company-general-info.component.html:15]: fixed

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.scss (hunk 13 above)
Comment (dhaval2989) [reply src/app/reusable/nv-company-general-info/nv-company-general-info.component.scss:69]: this is a variable defined in base.scss

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts (hunk 14 above)
Comment (dhaval2989) [reply src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts:16]: fixed

File: src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts (hunk 15 above)
Comment (dhaval2989) [reply src/app/reusable/nv-company-general-info/nv-company-general-info.component.ts:20]: fixed

File: src/app/reusable/nv-custom-datatable/nv-custom-datatable.component.scss (hunk 16 above)
Comment (dhaval2989) [reply src/app/reusable/nv-custom-datatable/nv-custom-datatable.component.scss:98]: fixed

File: src/app/reusable/user-list-table/user-list-table.component.scss (hunk 18 above)
Comment (dhaval2989) [reply src/app/reusable/user-list-table/user-list-table.component.scss:12]: fixed

File: src/styles.scss (hunk 19 above)
Comment (dhaval2989) [reply src/styles.scss:8]: this is not part of my component

File: src/app/main/components/dashboard/dashboard.component.html (hunk 21 above)
Comment (dhaval2989) [reply src/app/main/components/dashboard/dashboard.component.html:128]: fixed

File: src/app/main/components/dashboard/dashboard.component.ts (hunk 22 above)
Comment (dhaval2989) [reply src/app/main/components/dashboard/dashboard.component.ts:14]: this is not part of my component

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 25 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:8]: Yes working

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 26 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:22]: fixed

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 27 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:31]: fixed

File: src/app/main/components/user-list-view/user-list-view.component.scss (hunk 29 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.scss:36]: fixed

File: src/app/main/components/user-list-view/user-list-view.component.ts (hunk 30 above)
Comment (dhaval2989) [reply src/app/main/components/user-list-view/user-list-view.component.ts:5]: fixed

File: src/app/reusable/user-list-table/user-list-table.component.ts (hunk 31 above)
Comment (dhaval2989) [reply src/app/reusable/user-list-table/user-list-table.component.ts:122]: fixed

File: src/app/main/components/dashboard/dashboard.component.html (hunk 4 above)
Comment (dhaval2989) [reply src/app/main/components/dashboard/dashboard.component.html:21]: Its not part of my component.

**Pull request merged**